assets/src/design-system/components/typeahead/test/input.js

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+
+import { fireEvent } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { renderWithProviders } from '../../../testUtils/renderWithProviders';
+import TypeaheadInput from '../input';
+
+describe('Typeahead <Input />', () => {
+  const onClickMock = jest.fn();
+  const handleClearInputMock = jest.fn();
```

Comment:
If we don't already do this - I think it would be prudent to add a mock clear before each test to make sure there's not leakage between tests. If we add the following it should just work 🤷 

```
beforeEach(jest.clearAllMocks);
```